.generated-sources/blocklist-api/src/main.rs

```diff
@@ -0,0 +1,3 @@
+fn main() {
+    // Dummy main so the crate is valid before `build.rs` runs.
```

Comment:
So in my first commit I had a `build.rs` which functionally worked but was giving "unused code" warnings. 
I tried moving it to separate locations which didn't fix the issue, that's why I went with the regular crate option so I can pull the blocklist-client dependencies regularly. 
Maybe I didn't setup the `build.rs` properly.
Is there a strong preference to have the `build.rs` here?

Reply:
I might be wrong, but I don't think either `build.rs` or this file are actually needed.

Reply:
> So in my first commit I had a `build.rs` which functionally worked but was giving "unused code" warnings. I tried moving it to separate locations which didn't fix the issue, that's why I went with the regular crate option so I can pull the blocklist-client dependencies regularly. Maybe I didn't setup the `build.rs` properly. Is there a strong preference to have the `build.rs` here?

Okay, interesting. Was it the code in `build.rs` that triggered the warning or something else? I'd be curious to see the error.

Reply:
I think I see the same warning now. I checked out your commit `b28b5cbd1799ce668e1844ffa5977d4ed6f8a1d5` (kudos for maintaining a good commit history ⭐🚀) and built the blocklist client. One thing I'm noticing that strikes me as quite unconventional there is that you declare sub-modules in your `build.rs`. I suppose that's where the warning comes from, because the compiler thinks those are build-specific modules that contains a lot of functions and structs that are not used within `build.rs`.

Reply:
So I think you would not have that problem if you use `blocklist-client` as a build dependency for `.generated-sources/blocklist-api` and take what you have in `blocklist-openapi-gen/src/main.rs` and put that in a `.generated-sources/blocklist-api/build.rs` instead.

Reply:
I just did the same with checking out the commit. (somehow the CI results are missing).

I see, so we don't mind the separate directory under `.generated-sources` but instead have the `build.rs`.
Since I originally had `build.rs` inside the same blocklist crate, I couldn't access the dependencies through imports but I most likely forgot to try that even when I moved `build.rs` out to a different crate. 

I"ll try it now

Reply:
@netrome @djordon  I tried keeping only one crate (blocklist-api) by placing the openAPI spec files there as well and having the openAPI spec get generated via `build.rs`. The issue was that since the generated client code also gets placed there, it would overwrite the crate `Cargo.toml`, getting rid of the `build-dependencies` needed for `build.rs`.
I played around with post-execution file editing, but I found that to not be as clean so I went back to keeping two separate crates: One for the generated openAPI spec and another for the generated code. 

Let me know if that's acceptable or if I missed something in getting it to work by keeping only one crate.

Reply:
That's fine by me. I'll take another look just to be sure though.

Reply:
Okay interesting. I thought that would work basically the same as `.generated-sources/emily`, but I guess there are subtle details I'm missing here. Thank you for trying it out, and yeah I'm okay with having the two crates. If there are more angles to investigate, we can leave that for a follow-up as well. It doesn't have to be a blocker for this PR.